src/components/search/SearchBar.tsx

```diff
@@ -0,0 +1,180 @@
+  search?: Search;
+  nodes?: RenderNodeMap;
+  handleSearchChange?: HandleSearchChange;
+}
+
+export class SearchBar extends React.PureComponent<SearchStoreProps, {}> {
+  private handleInput(value: string) {
+    const nodes = this.findNodes(value);
+    this.props.handleSearchChange({
+      active: true,
+      value: value,
+      nodes: nodes,
+      selected: 0
+    });
+    this.dragBackground();
+  }
+
+  private getAllNodes() {
+    return Object.entries(this.props.nodes).map(item => {
+      return {
+        uuid: item[0],
+        data: item[1]
+      };
+    });
+  }
+  private findNodes(value: string) {
+    const nodes = this.getAllNodes();
+    return nodes.filter(item => {
+      const nodeItem: any = item.data.node.actions;
+      if (nodeItem.length > 0) {
+        //find by type
+        const foundTitle = i18n
+          .t(`actions.${nodeItem[0].type}.name`)
+          .toLocaleLowerCase()
+          .includes(value.toLocaleLowerCase());
+        //find by content
+        const foundContent = nodeItem[0].hasOwnProperty('text')
+          ? item.data.node.actions[0].text.includes(value)
+          : nodeItem[0].hasOwnProperty('result_name')
+          ? item.data.node.actions[0].result_name.includes(value)
+          : false;
+
+        return foundTitle || foundContent;
+      } else {
+        return false;
+      }
+    });
+  }
+
+  private dragBackground() {
+    const canvasBg = document.getElementById('panzoom');
+    if (this.props.search.nodes[this.props.search.selected]) {
+      const ui = this.props.search.nodes[this.props.search.selected].data.ui.position;
+      const width = window.innerWidth / 2;
+      const height = window.innerHeight / 2;
+      if (canvasBg) {
+        canvasBg.style.transform = `matrix(1, 0, 0, 1, ${width - ui.left}, ${height - ui.top})`;
+      }
+    }
+  }
+
+  private toggleMoveSelected(type: 'up' | 'down') {
+    switch (type) {
+      case 'up':
+        this.props.handleSearchChange({
+          active: true,
+          value: this.props.search.value,
+          nodes: this.props.search.nodes,
+          selected: this.props.search.selected - 1 < 0 ? 0 : this.props.search.selected - 1
+        });
+        break;
+      case 'down':
+        var down = this.props.search.selected - 1;
+        if (down < 0) {
+          down = 0;
+        }
+        this.props.handleSearchChange({
+          active: true,
+          value: this.props.search.value,
+          nodes: this.props.search.nodes,
+          selected:
+            this.props.search.selected < this.props.search.nodes.length - 1
+              ? this.props.search.selected + 1
+              : this.props.search.nodes.length - 1
+        });
+        break;
+    }
+    this.dragBackground();
+  }
+
+  private closeSearch() {
+    this.props.handleSearchChange({
+      active: false,
+      value: this.props.search.value,
+      nodes: this.props.search.nodes,
+      selected: 0
+    });
+  }
```

Comment:
You could destructure this.props.search in this function too:
```suggestion
  private closeSearch() {
    const { value, nodes } = this.props.search;
  
    this.props.handleSearchChange({
      active: false,
      value,
      nodes,
      selected: 0
    });
  }
```